docs/guide.md

```diff
@@ -0,0 +1,69 @@
+program without having to rewrite, debug, or verify.
+
+Like other languages, the Dafny programming language and tool set includes
+libraries that provide functionality in various domains. This document describes 
+the _Dafny Core Library_. Dafny programs are copmponed of modules; to ensure
+verification soundness, Dafny imposes a strict non-circular dependency relationship
+among modules. This Core library contains functionality that may be needed
+by other libraries, but it itself is standalone.
+
+All the supported, delivered libraries will
+- be stable --- a minimum of backwards incompatible changes
+- be verified --- proved to meet specifications and tested with expected use cases,
+with each release of the library and of the Dafny toolset
+- be efficient --- efficient in both proofs and executables that use the libraries
+
+The Dafny code for the library is written to be understandable with a minimum of inline documentation
+and is open source and available [on github](https://github.com/dafny-lang/libraries). This document is
+meant to be tutorial and example-based.
+
+## Content of the Core Library
+
+The sections below describe how to use each of the principal modules of the library:
+- [Dafny.Boxes](#sec-boxes) -- containers that enable encapsulaton of success and failure results in a single datatype
+- [Bafny.BoundedInts](#sec-boundedints) -- definitions of types and cnostants for bounded integer arithmetic
+- [Dafny.FileIO}(#sec-fileio} -- basic input and output operations
+- [Dafny.Math](#sec-math) --- basic mathematical operations and lemmas
+- [Dafny.Unicode](#sec-unicode) --- tools to manipulate unicode characters and strings
+- [Dafny.Collections](#sec-collections) -- functions and lemmas about sequences, sets and maps
+- [Dafny.NonlinearArith](sec-nonlinear-arith) --  lemmas and functions for non-linear arithmetic
+- [Dafny.Relations](#sec-relations) -- predicates and lemmas about properties of functions
+
+
+## How to use the library
+
+One uses the library on the command-line by including a path to the library in the `--library` option.
+
+At present, you must have a copy of the library in your file system to use the library. Either
+- obtain a release zip file from [the library github release page](https://github.com/dafny-lang/libraries/releases) and unzip it in a folder of your choice, or
+- clone the library repository (`git clone https://github.com/dafny-lang/libraries.git`) to a location of your choice
+
+In either case, the relevant directory to point the `--library` option to is named `dafny`, either at the top of the zip file or at `libraries/src/dafny` in the clone.
+
+In VSCode, TBD
+
+
+---
+{% include_relative Boxes.md %}
+---
+{% include_relative BoundedInts.md %}
+---
+{% include_relative FileIO.md %}
+---
+{% include_relative Math.md %}
+---
+{% include_relative Collections.md %}
+---
+{%include_relative Relations.md %}
+---
+{% include_realtive Unicode.md %}
+---
+{% include_relative NonlinearArithmetic.md %}
```

Comment:
Could we move these to sit within their respective subdirectories instead?

Reply:
I'll try, but I'm quite sure that github pages does not support rendering pages outside the directory you point it to (in this case 'docs')

Reply:
Got this reorg, though through regular links and not include_relative